src/sentry/integrations/slack/link_team.py

```diff
@@ -0,0 +1,199 @@
+        super().__init__(*args, **kwargs)
+
+        self.fields["team"].choices = [(team.id, team.slug) for team in teams]
+        self.fields["team"].widget.choices = self.fields["team"].choices
+
+
+class SlackLinkTeamView(BaseView):
+    def get_identity(self, integration, slack_id):
+        try:
+            idp = IdentityProvider.objects.get(type="slack", external_id=integration.external_id)
+        except IdentityProvider.DoesNotExist:
+            logger.error(
+                "slack.action.invalid-team-id", extra={"slack_id": integration.external_id}
+            )
+            return self.respond(status=403)
+
+        try:
+            identity = Identity.objects.select_related("user").get(idp=idp, external_id=slack_id)
+        except Identity.DoesNotExist:
+            # I don't think this could be possible but just in case
+            return self.respond(status=403)
+        return identity
+
+    def send_slack_message(self, request, client, token, text, channel_id, integration):
+        payload = {
+            "token": token,
+            "channel": channel_id,
+            "text": text["body"],
+        }
+        headers = {"Authorization": "Bearer %s" % token}
+        try:
+            client.post("/chat.postMessage", headers=headers, data=payload, json=True)
+        except ApiError as e:
+            message = str(e)
+            if message != "Expired url":
+                logger.error("slack.link-notify.response-error", extra={"error": message})
+        else:
+            return render_to_response(
+                "sentry/slack-post-linked-team.html",
+                request=request,
+                context={
+                    "heading_text": text["heading"],
+                    "body_text": text["body"],
+                    "channel_id": channel_id,
+                    "team_id": integration.external_id,
+                },
+            )
+
+    @transaction_start("SlackLinkTeamView")
+    @never_cache
+    def handle(self, request, signed_params):
+        params = unsign(signed_params)
+        integration = Integration.objects.get(id=params["integration_id"])
+        teams = Team.objects.filter(
+            organization__in=integration.organizations.all(), status=TeamStatus.VISIBLE
+        ).order_by("slug")
+        channel_name = params["channel_name"]
+        channel_id = params["channel_id"]
+        form = SelectTeamForm(teams, request.POST or None)
+        if form.is_valid():
+            team_id = form.cleaned_data["team"]
+
+        if request.method != "POST":
+            return self.respond(
+                "sentry/slack-link-team.html",
+                {
+                    "form": form,
+                    "teams": teams,
+                    "channel_name": channel_name,
+                    "provider": integration.get_provider(),
+                },
+            )
+
+        team = Team.objects.get(id=team_id)
+        INSUFFICIENT_ROLE_MESSAGE = {
```

Comment:
Move these strings from the function to the top of the file.

Reply:
I don't have access to the variables used to fill in the f-string until this point though

Reply:
I can do a workaround using `.format()` but I thought we as an organization decided to use f-strings, and I'm not sure I can accomplish this with one.

Reply:
Ok let's leave them here.